contracts/AlphaStakingV3.sol

```diff
@@ -0,0 +1,169 @@
+  using SafeMath for uint;
+
+  event SetWorker(address worker);
+  event Stake(address owner, uint share, uint amount);
+  event Unbond(address owner, uint unbondTime, uint unbondShare);
+  event Withdraw(address owner, uint withdrawShare, uint withdrawAmount);
+  event CancelUnbond(address owner, uint unbondTime, uint unbondShare);
+  event Reward(address worker, uint rewardAmount);
+  event Extract(address governor, uint extractAmount);
+  event SetMerkle(address merkle);
+
+  uint public constant STATUS_READY = 0;
+  uint public constant STATUS_UNBONDING = 1;
+  uint public constant UNBONDING_DURATION = 30 days;
+  uint public constant WITHDRAW_DURATION = 3 days;
+
+  struct Data {
+    uint status;
+    uint share;
+    uint unbondTime;
+    uint unbondShare;
+  }
+
+  IERC20 public alpha;
+  address public governor;
+  address public pendingGovernor;
+  address public worker;
+  uint public totalAlpha;
+  uint public totalShare;
+  mapping(address => Data) public users;
+  address public merkle;
+
+  modifier onlyGov() {
+    require(msg.sender == governor, 'onlyGov/not-governor');
+    _;
+  }
+
+  modifier onlyWorker() {
+    require(msg.sender == worker || msg.sender == governor, 'onlyWorker/not-worker');
+    _;
+  }
+
+  modifier onlyMerkle() {
+    require(msg.sender == merkle, 'onlyMerkle/not-merkle');
+    _;
+  }
+
+  function initialize(IERC20 _alpha, address _governor) external initializer {
+    alpha = _alpha;
+    governor = _governor;
+  }
+
+  function setWorker(address _worker) external onlyGov {
+    worker = _worker;
+    emit SetWorker(_worker);
+  }
+
+  function setPendingGovernor(address _pendingGovernor) external onlyGov {
+    pendingGovernor = _pendingGovernor;
+  }
+
+  function setMerkle(address _merkle) external onlyGov {
+    merkle = _merkle;
+    emit SetMerkle(_merkle);
+  }
+
+  function acceptGovernor() external {
+    require(msg.sender == pendingGovernor, 'acceptGovernor/not-pending');
+    pendingGovernor = address(0);
+    governor = msg.sender;
+  }
+
+  function getStakeValue(address user) external view returns (uint) {
+    uint share = users[user].share;
+    return share == 0 ? 0 : share.mul(totalAlpha).div(totalShare);
+  }
+
+  function stake(uint amount) external nonReentrant {
+    _stake(msg.sender, amount);
+  }
+
+  function _stake(address owner, uint amount) internal returns (uint share) {
+    require(amount >= 1e18, 'stake/amount-too-small');
+    Data storage data = users[owner];
+    if (data.status != STATUS_READY) {
+      emit CancelUnbond(owner, data.unbondTime, data.unbondShare);
+      data.status = STATUS_READY;
+      data.unbondTime = 0;
+      data.unbondShare = 0;
+    }
+    alpha.safeTransferFrom(msg.sender, address(this), amount);
+    share = totalAlpha == 0 ? amount : amount.mul(totalShare).div(totalAlpha);
+    totalAlpha = totalAlpha.add(amount);
+    totalShare = totalShare.add(share);
+    data.share = data.share.add(share);
+    emit Stake(owner, share, amount);
```

Comment:
Let's separate normal Stake and StakerFor events, so we can separately track staking from claims in Staking contract.

Reply:
Fixed